### PR TITLE
add isDevToolsOpen

### PR DESCRIPTION
**Pull request by jonlepage**
fix: https://github.com/nwjs/nw.js/issues/4487

### CODE REVIEW

**Comment by bmatusiak** [comment]
so, when i run 
```
chrome.windows.getAll({ populate: true, windowTypes:['devtools'] }, function( wins ) {
      return wins.length > 0;
    })
```

it returns `undefined`, so it seems the callback acts async
```
{
    "node": "21.1.0",
    "nw": "0.83.0",
    "nw-flavor": "sdk",
    "chromium": "120.0.6099.129"
}
```

File: src/resources/api_nw_newwin.js [hunk 1]
```diff
@@ -760,7 +760,11 @@ apiBridge.registerCustomHook(function(bindingsAPI) {
       callback(cwindows.map(create_nw_win));
     });
   });
-
+  apiFunctions.setHandleRequest('isDevToolsOpen', function() {
+    return chrome.windows.getAll({ populate: true, windowTypes:['devtools'] }, function( wins ) {
+      return wins.length > 0;
+    })
+  });
```
Comment (ayushmanchhabra) [inline src/resources/api_nw_newwin.js:767]:
```suggestion
  apiFunctions.setHandleRequest('isDevToolsOpen', function(callback) {
    return chrome.windows.getAll({ populate: true, windowTypes: ['devtools'] }, function(wins) {
      callback(wins.length > 0);
    })
  });
```

Comment (ayushmanchhabra) [reply src/resources/api_nw_newwin.js:767]: I think this should fix the async issue. Example usage (assuming this is how the API was implemented pre v13):

```js
nw.Window.isDevToolsOpen((status) => {
  console.log(status)
});
```

Comment (ayushmanchhabra) [reply src/resources/api_nw_newwin.js:767]: Tested locally and seems to work nicely:

Screenshot (16)

Comment (ayushmanchhabra) [reply src/resources/api_nw_newwin.js:767]: @rogerwang what are your thoughts?

Comment (rogerwang) [reply src/resources/api_nw_newwin.js:767]: Your fix is good. Thanks.

Comment (jonlepage) [reply src/resources/api_nw_newwin.js]: done, thank for your time friend

Comment (ayushmanchhabra) [reply src/resources/api_nw_newwin.js]: @rogerwang does a similar change need to be made on src/resources/nw_api_window.js?

**Comment by rogerwang** [comment]
Please fix the documentation as well.

**Comment by rogerwang** [comment]
no need to update nw_api_window.js ...

**Comment by jonlepage** [comment]
> > Please fix the documentation as well.
> 
> @jonlepage please remove mention of isDevToolsOpen from the [known issues](https://nwjs.readthedocs.io/en/latest/For%20Users/Migration/From%200.12%20to%200.13/#known-issues), thanks

Sorry, I don't understand. I don't see any trace of JSDoc for the documentation, and if you are referring to TypeScript documentation, it should be edited at https://github.com/DefinitelyTyped/DefinitelyTyped

**Comment by rogerwang** [comment]
https://github.com/nwjs/nw.js/blob/nw91/docs/References/Window.md

**Comment by ayushmanchhabra** [comment]
> @jonlepage please remove mention of isDevToolsOpen from the [known issues](https://nwjs.readthedocs.io/en/latest/For%20Users/Migration/From%200.12%20to%200.13/#known-issues)

https://github.com/nwjs/nw.js/blob/nw91/docs/For%20Users/Migration/From%200.12%20to%200.13.md

@jonlepage sorry should have linked the Markdown docs

**Comment by jonlepage** [comment]
done, sry for the time, I was a bit lost because I no longer had the fork on my server. So, I used the one from GitHub, which threw me off a bit for navigation. But it should be fine now. :)

you can see he have some bug ! and sometime file not appear ~ congrat Microsoft :)🙄
image

File: docs/For Users/Migration/From 0.12 to 0.13.md [hunk 2]
```diff
@@ -59,7 +59,7 @@
 ### Known issues
 
 + The following window options passed to `nw.Window.open()` is not effective on **Linux**: `resizable` for now; try to set them in the callback.
-+ `reloadDev()` and `isDevToolsOpen()` of `nw.Window` are not supported for now
++ `reloadDev()` of `nw.Window` are not supported for now
```
[the remark this answers is not in the document]
Comment (ayushmanchhabra) [reply docs/For Users/Migration/From 0.12 to 0.13.md:62]: One last nit and should be good to go

**Comment by ayushmanchhabra** [comment]
@rogerwang this looks good to me, let us know if any other changes are required

**Pull request merged**